### PR TITLE
Add Feature Expand and collapse in FMCARD

### PR DESCRIPTION
**Pull request by muhammad-farhan64**
### Detailed description of changes

1. add feature expand and collapse in FMCARD
### Recommendations on how to test the changes

1. go to storybook
2.  set isCollapseAble true

### Checklist for developer

- [x] The code builds clean without any errors or warnings
- [x] The code does not contain commented out code
- [x] The code does not log anything to console
- [x] I have added unit test(s) to cover new code and successfully executed ran it
- [x] I have thoroughly tested the new code and any adjacent features it may affect

### Link to related issue(s)

...

### FIGMA

Screen Shot 2022-02-17 at 17 58 46
Screen Shot 2022-02-17 at 17 58 58

Storybook
...

https://user-images.githubusercontent.com/74447557/154467858-385ab87f-24c1-4c00-ab3f-b0998e84c513.mp4

### CODE REVIEW

File: src/Components/Molecules/FMCard/FMCard.interfaces.tsx [hunk 1]
```diff
@@ -5,4 +5,5 @@ export interface IFMCard extends CardProps {
   testID: `card-${string}`;
   actions?: ReactChild;
   title?: string;
+  isCollapseAble?: boolean;
```
Comment (ardiwine) [inline src/Components/Molecules/FMCard/FMCard.interfaces.tsx:8]: isCollapsible

File: src/Components/Molecules/FMCard/index.tsx [hunk 2]
```diff
@@ -1,31 +1,66 @@
...
+import { FAIcon } from "@Atoms/Icon";
 
 export const FMCard = memo((props: IFMCard) => {
-  const { testID, actions, title, children, ...materialUIProps } = props;
+  const {
+    testID,
+    actions,
+    title,
+    children,
+    isCollapseAble,
```
Comment (ardiwine) [inline src/Components/Molecules/FMCard/index.tsx:17]: isCollapsible

**Review by ardiwine — CHANGES_REQUESTED** [review]

File: src/Components/Molecules/FMCard/index.tsx [hunk 3]
```diff
@@ -1,31 +1,66 @@
-import React, { memo } from "react";
+import React, { memo, useState } from "react";
 import Card from "@mui/material/Card";
 import CardActions from "@mui/material/CardActions";
 import CardContent from "@mui/material/CardContent";
 
 import { IFMCard } from "./FMCard.interfaces";
 import { FATypography } from "@Atoms/FATypography";
+import { Box, Collapse } from "@mui/material";
```
Comment (bryantobing12) [inline src/Components/Molecules/FMCard/index.tsx:8]: import nya kalau boleh di update farhan. `Box` dari `@mui/material/box`, `collapse` juga sama dan dinaikin keatas sesuai order.

File: src/Components/Molecules/FMCard/index.tsx [hunk 4]
```diff
@@ -1,31 +1,66 @@
...
+  const [isOpen, setIsOpen] = useState(true);
   return (
     <Card data-testid={testID} {...materialUIProps}>
       <CardContent>
         {title && (
-          <>
+          <Box
+            sx={{
+              display: "flex",
+              justifyContent: "space-between",
```
Comment (bryantobing12) [inline src/Components/Molecules/FMCard/index.tsx:30]: tambahin `align-items`  center biar title dan icon nya sejajar

File: src/Components/Molecules/FMCard/index.tsx [hunk 5]
```diff
@@ -1,31 +1,66 @@
...
               variant="h6"
               fontWeight={700}
-              sx={{ marginBottom: 3.75 }}
             >
               {title}
             </FATypography>
-          </>
+            {isCollapsible && (
+              <Box>
+                <FAIcon
```
Comment (bryantobing12) [inline src/Components/Molecules/FMCard/index.tsx:43]: kayaknya better pake button icon biar bisa navigasi dari keyboard untuk a11y

File: src/Components/Molecules/FMCard/index.tsx [hunk 6]
```diff
@@ -1,31 +1,67 @@
...
+                  onClick={() => {
+                    setIsOpen(!isOpen);
+                  }}
+                  style={{ cursor: "pointer" }}
```
Comment (bryantobing12) [inline src/Components/Molecules/FMCard/index.tsx:47]: kalau bisa `sx` lebih bagus daripada inline style. tapi kalau diganti jadi button ga perlu cursor pointer lagi

File: src/Components/Molecules/FMCard/index.tsx [hunk 7]
```diff
@@ -1,31 +1,67 @@
...
             <FATypography
               testID="typography-card"
               variant="h6"
               fontWeight={700}
-              sx={{ marginBottom: 3.75 }}
             >
               {title}
             </FATypography>
-          </>
+            {isCollapsible && (
```
Comment (bryantobing12) [inline src/Components/Molecules/FMCard/index.tsx:41]:
```suggestion
{isCollapsible && (
      <FAIconButton
        testID="icon-button-arrow-collapse"
        onClick={() => {
          setIsOpen(!isOpen);
        }}
      >
        <FAIcon testID="icon-arrow-collapse">arrow_drop_down</FAIcon>
      </FAIconButton>
)}
```

File: src/Components/Molecules/FMCard/index.tsx (hunk 6 above)
Comment (muhammad-farhan64) [reply src/Components/Molecules/FMCard/index.tsx:47]: oh oke

**Review by bryantobing12 — APPROVED** [review]

**Review by bryantobing12 — APPROVED** [review]

**Review by didikmulyadi — APPROVED** [review]

**Pull request merged**